File: NOV29/factRecursion.jl
#!/usr/bin/julia

function fact(n)
    if ( n < 0 )
      return -1
    elseif ( n == 0 || n == 1 )
      return 1
    else
      return n * fact(n-1)
    end
 end
 
 # Here is the main program, note I have to convert the
 # string command line argument to an Int64
 
 println(fact(parse(Int64,ARGS[1])))
 exit(0)

File: NOV29/factRecursion.py
#!/usr/bin/python3

import sys

def factorial(my_val):
    if (my_val < 0):
        return "-- ERROR --\n"
    elif(my_val == 0 or my_val == 1):
        return 1
    else:
        fact = 1
        return my_val * factorial(my_val-1)

print(factorial(int(sys.argv[1])))